lib/plugins/aws/package/compile/events/s3/index.js

```diff
@@ -126,14 +165,59 @@ class AwsCompileS3Events {
     // iterate over all buckets to be created
     // and compile them to CF resources
     _.forEach(bucketsLambdaConfigurations, (bucketLambdaConfiguration, bucketName) => {
+      let bucketProps = null;
+      if (bucketsLogicalIds[bucketName]) {
+        const providerBucket = provderS3[bucketsLogicalIds[bucketName].bucket];
+        if (providerBucket) {
+          bucketProps = Object.keys(providerBucket).reduce((acc, key) => {
+            if (key !== 'name') {
+              const property = _.upperFirst(key);
+              const allowedProperety = this.allowedBucketProperties[key];
+              if (!allowedProperety) {
+                const errorMessage = [
+                  `"${key}" is not a valid bucket property.`,
+                  'A bucket could only be configured with the following properties:\n',
+                  ['name'].concat(_.keys(this.allowedBucketProperties)).join(', '),
+                ].join('');
+                throw new this.serverless.classes.Error(errorMessage);
```

Comment:
I would not introduce such validation here, as we should have such validation solved globally for whole configuration (there's a dedicated issue: https://github.com/serverless/serverless/issues/6562 )

So maybe just accept recognized properties, ignore others, and do not do type checking (it's covered already by CF validation which is done prior deployment)